packages/core/http/core-http-server/src/versioning/types.ts

```diff
@@ -159,16 +156,16 @@ export interface VersionedRouter<Ctx extends RqCtx = RqCtx> {
 export type VersionedRouteRequestValidation<P, Q, B> = RouteValidatorFullConfig<P, Q, B>;
 
 /** @experimental */
-export interface VersionedRouteResponseValidation<R> {
-  [statusCode: number]: { body: RouteValidationFunction<R> | Type<R> };
+export interface VersionedRouteResponseValidation {
```

Comment:
It was a cool idea, but in practice it looks like inferring a generic (`R`) as union of types from a number-indexed object is too tall of a task for TS atm.